int32 sent_flag
---
int32 sent_flag_feedback
